src/rest.ts

```diff
@@ -39,18 +39,29 @@ export function getTDPDatabases(): Promise<IDatabaseDesc[]> {
   return getAPIJSON(`${REST_DB_NAMESPACE}/`);
 }
 
-export interface IViewDesc {
+export interface IViewDescription {
+
+  /**
+   * idType of the DBView, can be null
+   */
+  idType: string | null;
+
+  /**
+   * list of columns within this view, can be empty
+   */
+  columns: IServerColumn[];
+}
+
+export interface IDatabaseViewDesc extends IViewDescription {
```

Comment:
should we make the attributes in `IViewDescription` optional again like it was in the original version? not sure if we catch all potential problems in each plugin. making them optional again should be backwards compatible. if so it would be `... extends Partial<IViewDescription>`

Reply:
Yes, updated the definition.